manifests/job/present.pp

```diff
@@ -29,12 +33,43 @@
   include jenkins::cli
   include jenkins::cli::reload
 
+  if $temp_config_path != undef and $config != undef {
+    fail('You cannot set both $temp_config_path AND $config param, only one is required')
+  }
+
   if $jenkins::service_ensure == 'stopped' or $jenkins::service_ensure == false {
     fail('Management of Jenkins jobs requires \$jenkins::service_ensure to be set to \'running\'')
   }
 
   $jenkins_cli        = $jenkins::cli::cmd
-  $tmp_config_path    = "/tmp/${jobname}-config.xml"
+  if $config == undef {
+    $tmp_config_path = $temp_config_path
```

Comment:
I feel that use two variable names that are so similar is asking for trouble down the road...